manubot/cite/citekey.py

```diff
@@ -71,6 +40,68 @@ def standardize_citekey(citekey, warn_if_changed=False):
     return standard_citekey
 
 
+# NOTE: tentatively - must split to individual functions correspnding to 
+#       methods of CSL_dict class 
+def citekey_to_csl_item(citekey, prune=True):
+    """
+    Generate a CSL Item (Python dictionary) for the input citekey.
+    """
+    citekey == standardize_citekey(citekey, warn_if_changed=True)
+    source, identifier = citekey.split(':', 1)
+
+    if source in citeproc_retrievers:
+        citeproc_retriever = import_function(citeproc_retrievers[source])
+        csl_item = citeproc_retriever(identifier)
+    else:
+        msg = f'Unsupported citation source {source!r} in {citekey!r}'
+        raise ValueError(msg)
+
+    from manubot import __version__ as manubot_version
+    from manubot.cite.citeproc import (
+        csl_item_passthrough,
+        append_to_csl_item_note,
+    )
+
+    note_text = f'This CSL JSON Item was automatically generated by Manubot v{manubot_version} using citation-by-identifier.'
+    note_dict = {
+        'standard_id': citekey,
+    }
+    append_to_csl_item_note(csl_item, note_text, note_dict)
+
+    short_citekey = shorten_citekey(citekey)
+    csl_item = csl_item_passthrough(csl_item, set_id=short_citekey, prune=prune)
+
+    return csl_item
+
+import functools
+import logging
+import re
+
+# FIXME: this goes away with new classes
```

Comment:
I'd prefer to remove the `FIXME` statements that refer the proposed class architecture. Helpful to keep a personal record, but best to keep it out of the master code, since this is currently a proposal with an unknown outcome and implementation timeframe that is not required for the current code to function. Let's use `TODO`/`FIXME` when there are actual problems with the code that we are aware of but do not want to currently address.

Reply:
Ughh.

Reply:
Removed the comments to keep the master cleaner, but the idea in between the lines is the following: a combination of hardcoded tests that fails on forks (#140), cross-referencing to submodule (#145) are heavy toll on modularity and speed of developement. Hope I can propose some changes that make code more maintainable.